simulator/src/ledger/dumper.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(unused_must_use)]
```

Comment:
Minor: can we remove this warning suppression? I didn't see variables that is unused.

Reply:
Yes we can - the warning actually is caused no by unused variables, but by the fact that writeln! has a return value and so the compiler warns that you need to read that variable (we really don't need to). The alternative is to use `let _ = writelin!(..` . I can goi ahead and do that if you'd like

Reply:
I think this is because the return is a `Result` which may indicate a failure, especially when you're dealing with TCP stream.

The suggested use from the rust lang doc is to add `?` after `writeln!`. https://doc.rust-lang.org/std/macro.writeln.html

Reply:
In our case `.map_err(Error::IOError)?`

Reply:
Yeah, but the TCP failure would be from a closed or broken client connection which is handled upstream. But yeah - I'll go ahead and add that in

Reply:
Thanks!

Reply:
Done